storey/dtypes.py

```diff
@@ -190,29 +192,116 @@ def __init__(self, delay_in_seconds=0, emission_type=EmissionType.All):
         self.delay_in_seconds = delay_in_seconds
         EmitBase.__init__(self, emission_type)
 
+    @staticmethod
+    def name():
+        return 'afterPeriod'
+
 
 class EmitAfterWindow(EmitBase):
     def __init__(self, delay_in_seconds=0, emission_type=EmissionType.All):
         self.delay_in_seconds = delay_in_seconds
         EmitBase.__init__(self, emission_type)
 
+    @staticmethod
+    def name():
+        return 'afterWindow'
+
 
 class EmitAfterMaxEvent(EmitBase):
     def __init__(self, max_events, emission_type=EmissionType.All):
         self.max_events = max_events
         EmitBase.__init__(self, emission_type)
 
+    @staticmethod
+    def name():
+        return 'maxEvents'
+
 
 class EmitAfterDelay(EmitBase):
     def __init__(self, delay_in_seconds, emission_type=EmissionType.All):
         self.delay_in_seconds = delay_in_seconds
         EmitBase.__init__(self, emission_type)
 
+    @staticmethod
+    def name():
+        return 'afterDelay'
+
 
 class EmitEveryEvent(EmitBase):
+    @staticmethod
+    def name():
+        return 'everyEvent'
+
     pass
 
 
+def _dict_to_emit_policy(policy_dict):
+    mode = policy_dict.pop('mode')
+    if mode == EmitEveryEvent.name():
+        policy = EmitEveryEvent()
+    elif mode == EmitAfterMaxEvent.name():
+        if 'maxEvents' not in policy_dict:
+            raise ValueError('maxEvents parameter must be specified for maxEvents emit policy')
+        policy = EmitAfterMaxEvent(policy_dict.pop('maxEvents'))
+    elif mode == EmitAfterDelay.name():
+        if 'delay' not in policy_dict:
+            raise ValueError('delay parameter must be specified for afterDelay emit policy')
+
+        policy = EmitAfterDelay(policy_dict.pop('delay'))
+    elif mode == EmitAfterWindow.name():
+        policy = EmitAfterWindow(delay_in_seconds=policy_dict.pop('delay', 0))
+    elif mode == EmitAfterPeriod.name():
+        policy = EmitAfterPeriod(delay_in_seconds=policy_dict.pop('delay', 0))
+    else:
+        raise TypeError(f'unsopperted emit policy type: {mode}')
+
+    if policy_dict:
+        raise ValueError(f'got unexpected arguments for emit policy: {policy_dict}')
```

Comment:
I think this one could use a test to verify.

Reply:
`test_emit_policy_wrong_args`